TrackletAlgorithm/ProjectionRouter.h

```diff
@@ -129,7 +138,7 @@ void ProjectionRouter(BXType bx,
         auto nbins = LAYER!=0 ? nvmmelayers[LAYER-1] : nvmmedisks[DISK-1];
 
         // routing
-        ap_uint<3> iphi = (iphi5/(32/nvm))&(nbins-1);  // OPTIMIZE ME
+        ap_uint<3> iphi = (iphi5/((1<<nbits_maxvm)/nvm))&(nbins-1);  // OPTIMIZE ME
```

Comment:
this did not really get better. Is `nbins` a power of two such that the mask is all bits set? And why are you dividing by a power of two here rather than right-shifting like you do in other places? and is `nvm` a power of two or ...? I still think (if it works) a `ap_uint<>::range` would be a lot clearer here

Reply:
Agreed. It should be much clearer now. The resource usage before and after are identical (I actually hoped they might go down somewhat, but sadly it isn't so)

Reply:
Don't know about the resource usage, but splitting the computation of iphi (ivm in VMR) up and doing the nvm/32 at compile time, made the timing at lot better for the VMR: https://github.com/cms-tracklet/firmware-hls/blob/master/TrackletAlgorithm/VMRouter.h#L209 
Maybe that would help?

Reply:
@meisonlikesicecream That's good to know, but the code has since been rewritten so that the two divisions aren't needed.